mapper/software_management/examples/sm-agent.rs

```diff
@@ -0,0 +1,79 @@
+use mqtt_client::{Client, Config, Message, Topic};
+use software_management::message::*;
+use software_management::plugin::*;
+use software_management::plugin_manager::*;
+use software_management::software::*;
+
+use log::{debug, error, info};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let name = "sm-agent";
+    let operation_topic = Topic::new("tedge/sm-operations")?;
+    let status_topic = Topic::new("tedge/sm-status")?;
+    let error_topic = Topic::new("tedge/sm-errors")?;
+    let plugins = ExternalPlugins::open("/etc/tedge/sm-plugins")?;
+
+    env_logger::init();
+    info!("Starting SM-Agent");
+
+    let mqtt = Client::connect(name, &Config::default()).await?;
+    let mut errors = mqtt.subscribe_errors();
+    tokio::spawn(async move {
+        while let Some(error) = errors.next().await {
+            error!("{}", error);
+        }
+    });
+
```

Comment:
Before starting the request processing loop, we'll have to explicitly publish the initial software list to the cloud via the broker unless we expect the cloud mapper to send a message to the sm-agent to retrieve the software list that it needs to send to the cloud.